### PR TITLE
First attempt at Tropo migration

### PR DESCRIPTION
**Pull request by mheap**
> Please work with @leggetter to get this merged

## Description

This PR adds three new routes to Nexmo Developer:

* /migrate/tropo
* /migrate/tropo/sms
* /migrate/tropo/voice

These are't intended to be linked from the main site. Instead, they'll be linked to from a migration blog post and/or email campaign.

There are quite a few changes in this PR (43 changed files!). Here's an overview of them and why they happened

### New SMS/Voice building blocks created

These were added under `_examples/migrate/tropo` for a few reasons:

* Consistency - the blocks used show the same phone number / message as the Tropo blocks
* Isolation - our existing blocks don’t use the yaml format and are due an update soon. Using separate blocks means that we can’t accidentally break the guide whilst working on it
* Speed - honestly, it was the quickest way to get this done. We have some examples eg curl that there isn’t a Tropo equivalent for and it looked weird showing curl for Nexmo and node for Tropo. There’s not a way to exclude languages when rendering a set of blocks yet (probably a good feature to have)

### Navigation code updated

We needed a way to show expanded navigation for a section when the content we are showing doesn't live in the documentation. This allows us to expand messaging/sms when we're on the migrate/sms page

### Building Block filter

Whilst in most cases we want to show all prerequisites for creating an application, making a call etc it was too much to show when trying to contrast the amount of code needed between Nexmo and Tropo. I added a new `code_only` flag to `building_blocks_filter` that returns only the implementation code with no dependencies if set

### Other thoughts

There's a strange mix of HTML/Markdown in ERB files and the controllers. Ideally we'd want to be able to define this entire page as config (@martyndavies made a start on this for the podcast landing page), but time is short for this functionality and it renders correctly. Please review this PR as a consumer rather than as a developer, and we'll clean it up once we have support for configuration driven landing pages (once we have a tech lead)

## Deploy Notes

N/A

### CODE REVIEW

File: app/views/static/migrate.html.erb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<div class="container">
+  
+  <div class="row">
+    <div class="columns small-offset-2 small-8 center spacious">
+
+      <h1>Migrate from Tropo to Nexmo</h1>
+      <small>In case you have not seen it, please review our **migration program post-link here**,<br /> which includes free credit on our platform for Tropo customers.</small>
```
Comment (tbedford) [inline app/views/static/migrate.html.erb]: Don't forget to insert link here.

File: app/views/static/migrate.html.erb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        <br />
+        <div class="center">
+          <h1>Get Started</h1>
+
+          <p>Get started with the Nexmo platform in four steps:</p>
+          <br />
+        </div>
+
+        <h2>1. Register for a Nexmo account</h2>
+        <p>Sign up for a Nexmo account. Once you've signed up you'll receive €2 free credit which is enough to get started. We're offering a migration package to Tropo customers, including additional free credit. See our **INSERT BLOG POST LINK** migration blog post for more information</p>
```
Comment (tbedford) [inline app/views/static/migrate.html.erb]: Don't forget to insert link here.

File: app/views/static/migrate.html.erb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        </div>
+
+        <br />
+
+        <h2>2. Purchase a phone number</h2>
+        <p>Head over to the <a href="https://dashboard.nexmo.com/buy-numbers">Nexmo dashboard</a> and purchase a virtual number to use for your application. You may not need it if you're in a country that supports alpha senders - search for your destination country on the <a href="https://help.nexmo.com/hc/en-us">knowledge base</a> for more information. If you've already got a US number and would like to keep it, you can port the number to Nexmo <a href="https://dashboard.nexmo.com/keep-your-number">using the Nexmo dashboard</a>.</p>
+        <br />
+
+        <h2>3. Install the Nexmo Client Library</h2>
+        <p>Nexmo provides client libraries in six different languages! Whether you write Javascript, PHP, Python, Java, .net or Ruby, we've got you covered. Head over to the <a href="/tools">tools page</a> to learn more about what's available.</p>
```
Comment (tbedford) [inline app/views/static/migrate.html.erb]: '.net' -> '.NET'.

File: _examples/migrate/tropo/send-an-sms/nexmo/node [hunk 4]
```diff
@@ -0,0 +1,2 @@
+const text = 'Don't forget your meeting at 2 p.m. on Wednesday!'
```
Comment (lornajane) [inline _examples/migrate/tropo/send-an-sms/nexmo/node]: Needs double quotes or an escape character

**Comment by lornajane** [comment]
In a word of encouragement: I merged this branch in against Volta for a little local test and while there are some things to fix for sure, I don't think it's going to be awful.

**Pull request merged**